src/example-types/filters/tagsQuery.js

```diff
@@ -80,6 +80,43 @@ let filter = ({ tags, join, field, exact }) => ({
   },
 })
 
+let result = async (
+  { tags, join, field, exact, _meta, maxToCount = 20 },
+  search
+) => {
+  if (_.size(tags) > maxToCount) {
+    return null
+  }
+
+  let results = await Promise.all(
+    _.map(
+      async tag => ({
+        tag,
+        result: await search({
+          query: {
+            query_string: {
+              query: tagsToQueryString([tag], join),
+              default_operator: 'AND',
+              default_field:
+                field.replace('.untouched', '') + (exact ? '.exact' : ''),
+              ...(exact && { analyzer: 'exact' }),
+            },
+            ...(_meta.relevantFilters ? _meta.relevantFilters : {}),
+          },
+          size: 0,
+          track_total_hits: true,
+        }),
```

Comment:
This version of search is [curried](), which means this blob [is passed in as `aggs`](https://github.com/smartprocure/contexture-elasticsearch/blob/master/src/index.js#L36-L47). You can use this method by using the [filter](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filter-aggregation.html) or [filters agg](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filters-aggregation.html). Filters would be one query, filter would require running multiple.

If you want to do a straight search without aggregations, you'd need to run the search directly and join the filters with `bool` or `groupCombinator` manually. The path of least resistance is probably the filters agg, but it might make sense to test both options manually against elastic.

Reply:
@daedalus28 ty, that speeds me up and gets me thinking more in ES terms.

Reply:
@daedalus28 I think doing it with filter aggs will produce incorrect counts since I'm forced to pretend it's an exact match on tag.word which isn't what happens when you query on word and distance -- seems better for that reason to run a fresh query instead using the un-curried version of `search`. Correct me if I'm wrong here.

Reply:
also, supposing I'm right about that @daedalus28 , here's my next question: when I manage to get my hands on the un-curried search (node.getProvider().runSearch, I think) I'm going to be missing a bunch of potentially important stuff like `requestOptions` that was passed to the curried version. Don't I need to recover those arguments and pass them in again, and if so, how? or are they things that won't matter at this point?

Reply:
@daedalus28 check the new version when you get a chance -- that works perfectly on the first search you do on a fresh contexture init. However, on subsequent request the result fn doesn't get called and so you don't get counts when you add or remove a tag. Any concept what might cause that?